torch/jit/frontend.py

```diff
@@ -217,6 +217,35 @@ def remove_prefix(text, prefix):
     return aligned_prefix + aligned_suffix
 
 
+def check_fn_decorators_supported(fn_def, ctx):
+    """
+    Check that all decorators present in fn_def are supported by JIT.
+
+    Args:
+        fn_def: The ast.FunctionDef object.
+        ctx: A SourceContext instance. Used for source highlighting
+            when an unsupported decorator is found.
+    """
+    # Small function for converting AST subtrees representing decorators
+    # to their full names.
+    def get_decorator_name(decorator):
+        if isinstance(decorator, ast.Call):
+            return get_decorator_name(decorator.func)
+        elif isinstance(decorator, ast.Attribute):
+            return f"{get_decorator_name(decorator.value)}.{decorator.attr}"
+        elif isinstance(decorator, ast.Name):
+            return decorator.id
+        else:
+            raise RuntimeError(f"Unsupported node type: {type(decorator)}")
+
+    for decorator in fn_def.decorator_list:
+        if isinstance(decorator, ast.Call):
+            decorator_name = get_decorator_name(decorator)
+            if decorator_name == "torch.no_grad":
```

Comment:
Would this be safe against things like import aliasing?

import torch as to
from torch import no_grad

If it is not safe, could we fetch object according to name and compare against actual `torch.no_grad` object?

Reply:
Good point, I'm so used to seeing `@torch.no_grad` everywhere that I didn't realize. Yes, I think it is possible to fetch the object and compare against `torch.no_grad`.

Reply:
Done.